### ISSUE TITLE
Can't compile with tensorflow backend in the official circleci docker container

### ISSUE THREAD

**Issue by yukw777** [LeelaChessZero/lc0#908]
I tried to compile lc0 with tensorflow locally but could not get it to work, so I built an image using [the official circleci Docker file](https://github.com/LeelaChessZero/lc0/blob/master/.circleci/Dockerfile), and tried to build it on there, and I still see the same errors. I'm up-to-date with master with the following diff:
```
$ git diff
diff --git a/meson_options.txt b/meson_options.txt
index 872e2e8..5a99367 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -90,7 +90,7 @@ option('opencl',

 option('tensorflow',
        type: 'boolean',
-       value: false,
+       value: true,
        description: 'Enable TensorFlow backend')

 option('openblas',
```

Meson output:
```
The Meson build system
Version: 0.51.1
Source dir: /lc0
Build dir: /lc0/build/release
Build type: native build
DEPRECATION: Option uses prefix "build_", which is reserved for Meson. This will become an error in the future.
Project name: lc0
Project version: undefined
C++ compiler for the build machine: c++ (gcc 7.4.0 "c++ (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0")
C++ compiler for the host machine: c++ (gcc 7.4.0 "c++ (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0")
Build machine cpu family: x86_64
Build machine cpu: x86_64
Library libprotobuf found: NO
Found pkg-config: /usr/bin/pkg-config (0.29.1)
Found CMake: /usr/bin/cmake (3.10.2)
Run-time dependency protobuf found: NO (tried pkgconfig and cmake)
Program protoc found: YES (/usr/local/bin/protoc)

|Executing subproject protobuf method meson
|
|Project name: protobuf
|Project version: 3.5.1
|C++ compiler for the build machine: c++ (gcc 7.4.0 "c++ (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0")
|C++ compiler for the host machine: c++ (gcc 7.4.0 "c++ (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0")
|Compiler for C++ supports arguments -DHAVE_PTHREAD: YES
|Compiler for C++ supports arguments -Wno-sign-compare -Wsign-compare: YES
|Compiler for C++ supports arguments -Wno-unused-parameter -Wunused-parameter: YES
|Compiler for C++ supports arguments -Wno-ignored-qualifiers -Wignored-qualifiers: YES
|Compiler for C++ supports arguments /wd4146: NO
|Compiler for C++ supports arguments /wd4244: NO
|Compiler for C++ supports arguments /wd4305: NO
|Compiler for C++ supports arguments /wd4506: NO
|Run-time dependency threads found: YES
|Build targets in project: 6
|Subproject protobuf finished.

Library pthread found: YES
Library dl found: YES
Library libtensorflow_cc found: YES
Run-time dependency accelerate found: NO (tried pkgconfig and cmake)
Library mkl_rt found: YES
Library openblas.dll found: NO
Library openblas found: YES
Program ispc found: NO
Library OpenCL found: YES
Run-time dependency opencl found: YES 2.1
Library cublas found: YES
Library cudnn found: YES
Library cudart found: YES
Program nvcc found: YES (/usr/local/cuda/bin/nvcc)
Run-time dependency zlib found: YES 1.2.11
Run-time dependency GTest found: YES (building self)
Build targets in project: 14
Found ninja-1.8.2 at /usr/bin/ninja
```
Errors (same from multiple files):
```
...
/usr/local/include/tensorflow/bazel-genfiles/tensorflow/core/protobuf/config.pb.h: In member function 'tensorflow::GraphOptions* tensorflow::ConfigProto::mutable_graph_options()':
/usr/local/include/tensorflow/bazel-genfiles/tensorflow/core/protobuf/config.pb.h:4322:15: error: 'CreateMaybeMessage' was not declared in this scope
     auto* p = CreateMaybeMessage<::tensorflow::GraphOptions>(GetArenaNoVirtual());
               ^~~~~~~~~~~~~~~~~~
/usr/local/include/tensorflow/bazel-genfiles/tensorflow/core/protobuf/config.pb.h:4322:60: error: expected primary-expression before '>' token
     auto* p = CreateMaybeMessage<::tensorflow::GraphOptions>(GetArenaNoVirtual());
                                                            ^
/usr/local/include/tensorflow/bazel-genfiles/tensorflow/core/protobuf/config.pb.h: In member function 'tensorflow::RPCOptions* tensorflow::ConfigProto::release_rpc_options()':
/usr/local/include/tensorflow/bazel-genfiles/tensorflow/core/protobuf/config.pb.h:4386:65: error: no matching function for call to 'DuplicateIfNonNull(tensorflow::RPCOptions*&)'
     temp = ::google::protobuf::internal::DuplicateIfNonNull(temp);
                                                                 ^
...
```

**Comment by borg323** [comment]
Did you read tensorflow.md? The protobuf version used to build lc0 must match the one used for tensorflow. Also note that the tensorflow backend is not tested regularly and may have bit-rotted (but from a quick look looks fine).

**Comment by Ttl** [comment]
Tensorflow backend has not been updated to support folded batchnorm in convolution weights: https://github.com/LeelaChessZero/lc0/blob/488d7c241c9d2095225a13f9802b7627b2179b03/src/neural/network_tf.cc#L317-L318

So even if you get it to compile it won't work. Even if that is fixed it doesn't support any new neural net features used in the latest nets.

**Comment by yukw777** [comment]
@borg323 I did and the docker image in which I attempted to compile it does have the right protobuf version. It seems like as @Ttl this won't work even if I get it to compile though. Thanks everyone.

**Comment by Naphthalin** [comment]
@borg323 was anything relevant to tensorflow updated in the last 9 months, or is this still an issue?

**Comment by borg323** [comment]
The tensorflow backend was updated to work, but building it is still hard (unless meson finds libtensorflow_cc as a dependency which happens on some systems). The instructions in `tensorflow.md` are no longer relevant as we don't need `protoc` for the build any more. There is a pending pr to improve both issues (#1145), but it is very low priority.

**Comment by mooskagh** [comment]
I'm not sure whether it's still an issue.
Closing, please reopen if you still not able to build it.
